crates/ra_hir/src/expr.rs

```diff
@@ -792,40 +795,79 @@ impl ExprCollector {
             ast::ExprKind::Label(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::IndexExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::RangeExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
-            ast::ExprKind::MacroCall(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
+            ast::ExprKind::MacroCall(e) => {
+                let name = e.name().map(|nr| nr.as_name()).unwrap_or_else(Name::missing);
+
+                if let Some((module, _)) = self.resolver.module() {
+                    let def = module.find_macro(&name);
+
+                    if let Some(def_id) = def {
+                        let rules = db.macro_def(*def_id);
+
+                        if let Some(rules) = rules {
+                            let tt = e.token_tree().map(mbe::ast_to_token_tree);
+
+                            if let Some(Some((tt, _))) = tt {
+                                let result = rules.expand(&tt);
```

Comment:
Instead of expanding the macro manually, this should make use of `db.hir_parse` method.

The method needs a `HirFileId`, but you can turn a `MacroCallId` into `HirFileId`. To construct `MacroCalId`, you need to intern `MacroCallLoc`, which is constructed from `MacroDefId` and the location of macro expression.

This ... is confusing, see `Collector::collect_macro_expansion` function for an example of using macro infra